content/general/concepts/http/http.md

```diff
@@ -0,0 +1,107 @@
+---
+Title: 'HTTP'
+Description: 'HTTP is used for fetching HTML documents and other Web-based resources.'
+Subjects:
+  - 'Code Foundations'
+  - 'Web Development'
+Tags:
+  - 'Browsers'
+  - 'Development'
+  - 'Errors'
+  - 'HTTP'
+CatalogContent:
+  - 'learn-intermediate-javascript'
+  - 'paths/front-end-engineer-career-path'
+---
+
+Hypertext Transfer Protocol (HTTP) is used for fetching [HTML](https://www.codecademy.com/resources/docs/html) documents and other Web-based resources. It follows a model where a client (most commonly a browser) requests content stored somewhere on a server.
+
+## HTTP/HTTPS
+
+Instead of HTTP, a more secure version, HTTPS, should be used in order to encrypt information sent between the client and server. This encryption is done with Transport Layer Security (or TLS, formerly [SSL](https://www.codecademy.com/resources/docs/general/ssl)). Some benifits of HTTPS include:
+
+- Financial data, like credit card information or bank account numbers, is protected from interception.
+- Domain-ownership can be better verified by users.
+- A growing standard of trust around sites that use HTTPS.
+
+## Requests
+
+HTTP requests are stateless, meaning that all requests are independent and have no knowledge of one another. Requests contain the following parts:
+
+- The HTTP method being used (more information shortly).
+- The requested URL along with any queries or parameters.
+- The HTTP version, such as 1.1 or 2.0.
+- Any header information such as:
+  - The referer that tells the URL where the request came from.
+  - Any user agent information about the requesting client.
+  - A unique host name that is ideal for many pages on one server.
+  - Cookie data about the request.
+- A response-like body that contains the resource to be sent (common with the POST method).
+
+An HTTP client requests information specific URLs using four primary methods:
+
+| Method | Description                                                 |
+| ------ | ----------------------------------------------------------- |
+| GET    | Requests data, content, or other resources from the server. |
+| POST   | Sends data, content, or other resources to the server.      |
+| PUT    | Sends updates for existing content on the server.           |
+| DELETE | Deletes specific content from the server.                   |
+
+## Responses
+
+If the server is able to connect with the client and fullfill its request, it will send back a response that includes the following parts:
```

Comment:
```suggestion
If the server is able to connect with the client and fulfill its request, it will send back a response that includes the following parts:
```